src/class-row-action.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Duplicate Post class to manage the row actions.
+ *
+ * @package Duplicate_Post
+ */
+
+namespace Yoast\WP\Duplicate_Post;
+
+/**
+ * Represents the Row_Action class.
+ */
+class Row_Action {
+
+	/**
+	 * Holds the object to create the action link to duplicate.
+	 *
+	 * @var Link_Builder
+	 */
+	protected $link_builder;
+
+	/**
+	 * Initializes the class.
+	 *
+	 * @param Link_Builder $link_builder The link builder.
+	 */
+	public function __construct( Link_Builder $link_builder ) {
+		$this->link_builder = $link_builder;
+
+		$this->register_hooks();
+	}
+
+	/**
+	 * Adds hooks to integrate with WordPress.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_filter( 'post_row_actions', [ $this, 'add_action_link' ], 11, 2 );
+		\add_filter( 'page_row_actions', [ $this, 'add_action_link' ], 11, 2 );
+	}
+
+	/**
+	 * Hooks in the `post_row_actions` and `page_row_actions` filters to add a 'Rewrite & Republish' link.
+	 *
+	 * @param array    $actions The array of actions from the filter.
+	 * @param \WP_Post $post    The post object.
+	 *
+	 * @return mixed
+	 */
+	public function add_action_link( array $actions, \WP_Post $post ) {
+		$title = \_draft_or_post_title( $post );
+
+		if ( $post->post_status === 'publish' ) {
+			$actions['rewrite'] = '<a href="' . $this->link_builder->build_link( $post->ID ) .
```

Comment:
Should this URL be escaped by using `esc_url()`?
I noticed in other cases it is escaped, see for example in the method `add_rewrite_and_republish_post_button`

Reply:
It is already escaped since it's the output of `wp_nonce_url`, actually, so I'll remove the escaping in the other occurrence